torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3546,8 +3546,28 @@ struct to_ir {
           range, sv->asValue(val_range, method), subscript_exprs));
     }
     if (subscript_exprs[0].kind() == TK_SLICE_EXPR) {
-      return std::make_shared<SimpleValue>(emitBasicSlice(
-          range, sv->asValue(val_range, method), subscript_exprs));
+      if (sv->kind() == "module") {
```

Comment:
maybe add: `TODO @wconstab refactor using Symbol instead of string for comparison here`

and maybe some context on why we are intercepting the slicing for modulelists/sequentials here / normal code emission won't work. (doesn't have to be too long).